### PR TITLE
feat: [PLG-2808]: plans page new flow 

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
#### Screenshots

WE have updated the flow for Plans page , till now it was inside the subscriptions page and user were finding it diff to find it , now we have moved the Plans page out on the side Nav from where it will be directly visible below the Billing Page .

https://github.com/harness/harness-core-ui/assets/64455881/e5321b59-8d70-41e4-9215-46f1e03d6aa4

https://github.com/harness/harness-core-ui/assets/64455881/28235549-fe39-4447-b140-7e655f82234b

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by devinder-harness — DISMISSED** [review]

File: src/modules/27-platform/auth-settings/pages/subscriptions/SubscriptionTab.tsx [hunk 1]
```diff
@@ -151,9 +118,7 @@ const SubscriptionTab = ({
   return (
     <React.Fragment>
       {hasLicense && getBanner()}
-      <Layout.Horizontal className={css.subscriptionTabButtons} spacing="medium">
-        {accountData?.productLed && getSubscriptionTabButtons()}
-      </Layout.Horizontal>
+      <Layout.Horizontal className={css.subscriptionTabButtons} spacing="medium"></Layout.Horizontal>
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/auth-settings/pages/subscriptions/SubscriptionTab.tsx:121]: Empty Layout component? Doesn't have any children.

File: src/modules/27-platform/auth-settings/pages/subscriptions/SubscriptionsPage.tsx [hunk 2]
```diff
@@ -174,10 +174,26 @@ const SubscriptionsPage: React.FC = () => {
 
   function getModuleSelectElements(): React.ReactElement[] {
     return ACTIVE_MODULE_SELECT_CARDS.map(cardData => {
+      const currentUrl = window.location.href
+      const plansPage = false || currentUrl.includes('tab=PLANS') || currentUrl.includes('/plans')
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/auth-settings/pages/subscriptions/SubscriptionsPage.tsx:178]:
```suggestion
      const plansPage = currentUrl.includes('tab=PLANS') || currentUrl.includes('/plans')
```

File: src/modules/27-platform/auth-settings/pages/subscriptions/SubscriptionsPage.tsx [hunk 3]
```diff
@@ -221,6 +237,7 @@ const SubscriptionsPage: React.FC = () => {
     </Container>
   ) : (
     <SubscriptionTab
+      isPlansPage={false || window.location.href.includes('/plans')}
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/auth-settings/pages/subscriptions/SubscriptionsPage.tsx:240]:
```suggestion
      isPlansPage={window.location.href.includes('/plans')}
```

File: src/modules/10-common/components/AccountSideNav/AccountSideNav.tsx [hunk 4]
```diff
@@ -47,6 +47,7 @@ export default function AccountSideNav(): React.ReactElement {
       {NG_LICENSES_ENABLED && (
         <SidebarLink exact label={getString('common.subscriptions.title')} to={routes.toSubscriptions({ accountId })} />
       )}
+      {NG_LICENSES_ENABLED && <SidebarLink exact label={'Plans'} to={routes.toPlans({ accountId })} />}
```
Comment (mayankVermaHarness) [inline src/modules/10-common/components/AccountSideNav/AccountSideNav.tsx:50]: Use strings framework for `'Plans'`

**Review by mayankVermaHarness — CHANGES_REQUESTED** [review]
Added a few comments.

**Comment by mayankVermaHarness** [comment]
Will this need to be onboarded to Nav 2.0 (new Nav)?
CC: @avinashmadhwani02

**Review by devinder-harness — APPROVED** [review]

**Review by devinder-harness — APPROVED** [review]

File: src/modules/10-common/pages/SettingsPages/AccountSettingsPage.tsx [hunk 5]
```diff
@@ -417,10 +424,10 @@ export const AccountSettingsPage: React.FC = () => {
...
             />
             <SettingsResourceCard
-              label={<String stringID="common.billing" />}
-              id={SettingsResources.Billing}
-              icon={'nav-settings'}
-              route={routesV2.toBillingSettings({ accountId, module })}
+              label={<String stringID="common.subscriptions.tabs.plans" />}
+              id={SettingsResources.Plans}
+              icon={'filestore'}
+              route={routesV2.toPlans({ accountId, module })}
```
Comment (mayankVermaHarness) [inline src/modules/10-common/pages/SettingsPages/AccountSettingsPage.tsx:430]: All routes seem to be from RoutesV2 in this file. Intended?

File: src/modules/27-platform/auth-settings/RouteDestinations.tsx [hunk 6]
```diff
@@ -133,6 +132,14 @@ const AuthSettingsRoutes: React.FC = () => {
         >
           <Billing />
         </RouteWithLayout>
+        <RouteWithLayout
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/auth-settings/RouteDestinations.tsx:135]: Do we also need to add this to RouteDestinationsV2 i.e. `src/modules/27-platform/auth-settings/SettingsRouteDestinations.tsx`?

**Review by mayankVermaHarness — APPROVED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

**Review by devinder-harness — APPROVED** [review]

**Pull request merged**